assets/media-gallery.js

```diff
@@ -49,7 +49,7 @@ if (!customElements.get('media-gallery')) {
           if (this.elements.thumbnails) {
             activeMedia.parentElement.scrollTo({ left: activeMedia.offsetLeft });
           }
-          if (!this.elements.thumbnails || this.dataset.desktopLayout === 'stacked') {
+          if (!this.mql.matches || ((!this.elements.thumbnails || this.dataset.desktopLayout === 'stacked') && !this.dataset.hideVariantImages )) {
```

Comment:
This remove a behaviour that was there before. 
Right now on our published themes if you're layout is `stacked` and the variants are set to be `hidden` and you have thumbnails images, then it should scroll you into view the picture of the variant in question. 

I think what we want is: 
- if there are no thumbnails and variant images are hidden, then don't run the scroll into view part. 
- if there are no thumbnails and variant images are not hidden and there are more than one variant image, then run the scroll into view part
- if there are thumbnails and variant images and stacked, then scroll into view as well.

Reply:
Ok, JS strikes back, we have new changes that I think keep all default behaviours (desktop and mobile) but fix the jumpiness.
The logic we added bases the scrolling behaviour on the `section.settings.enable_sticky_info` and the media query.